arm_robots/scripts/calibration/calibrate_camera_to_mocap.py

```diff
@@ -0,0 +1,130 @@
+    parser.add_argument('camera_link_name', help='TF name of the camera link')
+    parser.add_argument('m', type=int, help='number of distinct data points (poses) to use in calibration')
+
+    args = parser.parse_args(rospy.myargv()[1:])
+
+    offsets = collect_calibration_data(args.camera_tf_name,
+                                       args.camera_link_name,
+                                       args.m)
+    print_calibration_results(offsets)
+
+
+def collect_calibration_data(camera_tf_name, camera_link_name, m):
+    """
+    takes a few measurements about where the camera is in mocap, where the calib board is in mocap, and where the
+    calib board is in the camera frame
+    """
+
+    tf = TF2Wrapper()
+    mocap_world_frame = 'mocap_world'
+    offsets = []
+    # NOTE: this is hard coded for the mocap+aruco calibration boards
+    fiducial_center_to_marker_corner = np.sqrt(0.118 ** 2 / 2)
+    fiducial_center_to_fiducial_mocap = transformations.compose_matrix(
+        translate=[-fiducial_center_to_marker_corner, fiducial_center_to_marker_corner, 0])
+    tag_idx = 0
+    camera2fiducial_last = None
+    for _ in trange(m):
+        # send TF from fiducial to mocap markers on the fiducial board
+        tf.send_transform_matrix(fiducial_center_to_fiducial_mocap,
+                                 parent=f"fiducial_{tag_idx}",
+                                 child=f"fiducial_{tag_idx}_mocap_markers")
+
+        while True:
+            # get the aruco transform once it is stable, and ensure it's different enough from the previous one
+            camera2fiducial = get_transform_stable(tf, camera_link_name, f"fiducial_{tag_idx}", 0.01)
+            if camera2fiducial_last is None:
+                break
+            dist = np.linalg.norm(camera2fiducial_last - camera2fiducial, ord='fro')
+            close = dist < 0.1
+            if not close:
+                break
+        camera2fiducial_last = camera2fiducial
+
+        fiducial2camera = transformations.inverse_matrix(camera2fiducial)
+
+        mocap2fiducial_markers = tf.get_transform(mocap_world_frame, f"mocap_calib{tag_idx}_calib{tag_idx}")
+        mocap2fiducial = mocap2fiducial_markers @ transformations.inverse_matrix(fiducial_center_to_fiducial_mocap)
+
+        mocap2camera_sensor_detected = mocap2fiducial @ fiducial2camera
+        mocap2camera_markers = tf.get_transform(mocap_world_frame, camera_tf_name)
+        mocap2camera_sensor_offset = np.linalg.solve(mocap2camera_markers, mocap2camera_sensor_detected)
+
+        offsets.append(mocap2camera_sensor_offset)
+    return offsets
+
+
+def print_calibration_results(offsets):
+    average_offset = average_transformation_matrices(offsets)
+    error = np.mean([np.linalg.norm(average_offset - t, ord='fro') for t in offsets])
+    trans = transformations.translation_from_matrix(average_offset)
+    rot = transformations.euler_from_matrix(average_offset)
+    roll, pitch, yaw = rot
+    print(Fore.GREEN)
```

Comment:
Created logger at the top but using print here; use logger instead?
Also might have to do a bit extra to get it working in a ROS environment (in my experience) such as:
```python
# at the top of the page
import rospy
rospy.init_node(..., log_level=rospy.INFO)
import importlib
importlib.reload(logging)

ch = logging.StreamHandler()
fh = logging.FileHandler(os.path.join(cfg.ROOT_DIR, "logs", "{}.log".format(datetime.now())))

logging.basicConfig(level=logging.INFO,
                    format='[%(levelname)s %(asctime)s %(pathname)s:%(lineno)d] %(message)s',
                    datefmt='%m-%d %H:%M:%S', handlers=[ch, fh])
```
alternatively remove the logger and logging import and stick to prints

Reply:
is this snippet worth putting somewhere common?

Reply:
what is the `cfg` here?